openedx/adg/lms/applications/helpers.py

```diff
@@ -285,3 +302,127 @@ def has_admin_permissions(user):
 
     is_user_admin = user.is_superuser or is_adg_admin(user) or BusinessLine.is_user_business_line_admin(user)
     return user.is_staff and is_user_admin
+
+
+def get_course_card_information(user, courses):
+    """
+    Decides status, grade and message for each course in the list and appends a dictionary containing status, grade,
+    message and course itself in a list
+
+    Arguments:
+        user: User object
+        courses: list containing Course objects
+
+    Returns:
+        list: A list containing dictionaries with status, grade, message and course
+    """
+    course_cards_information = []
+    is_any_course_started = False
+    is_locked = False
+
+    for course in courses:
+        message = grade = status = ''
+        pre_requisite_courses = get_prerequisite_courses_display(course)
+
+        if pre_requisite_courses:
+            message = LOCKED_COURSE_MESSAGE
+
+            for prereq in pre_requisite_courses:
+                course_grade = CourseGradeFactory().read(user, course_key=prereq['key'])
+
+                if not (course_grade and course_grade.passed):
+                    is_locked = True
+                    status = 'Locked'
```

Comment:
Can we please localize it if it is being displayed on frontend? Same goes for other values such as In Progress, Complete etc.